felix/iptables/table.go

```diff
@@ -677,7 +678,8 @@ func (t *Table) loadDataplaneState() {
 		if !t.ourChainsRegexp.MatchString(chainName) {
 			// Not one of our chains so it may be one that we're inserting rules into.
 			insertedRules := t.chainToInsertedRules[chainName]
-			if len(insertedRules) == 0 {
+			appendedRules := t.chainToAppendedRules[chainName]
+			if len(insertedRules) == 0 && len(appendedRules) == 0 {
 				// This chain shouldn't have any inserts, make sure that's the
 				// case.  This case also covers the case where a chain was removed,
 				// making dpHashes nil.
```

Comment:
Might want to change `dataplaneHasInserts` to `chainHasCalicoRule`? IIUC this could either be an insert or an append.